io-engine/src/subsys/nvmf/admin_cmd.rs

```diff
@@ -1,167 +1,168 @@
-extern "C" fn nvmf_create_snapshot_hdlr(req: *mut spdk_nvmf_request) -> i32 {
-    debug!("nvmf_create_snapshot_hdlr {:?}", req);
-
-    let subsys = unsafe { spdk_nvmf_request_get_subsystem(req) };
-    if subsys.is_null() {
-        debug!("subsystem is null");
-        return -1;
-    }
-
-    /* Only process this request if it has exactly one namespace */
-    if unsafe { spdk_nvmf_subsystem_get_max_nsid(subsys) } != 1 {
-        debug!("multiple namespaces");
-        return -1;
-    }
-
-    /* Forward to first namespace if it supports NVME admin commands */
-    let mut bdev: *mut spdk_bdev = std::ptr::null_mut();
-    let mut desc: *mut spdk_bdev_desc = std::ptr::null_mut();
-    let mut ch: *mut spdk_io_channel = std::ptr::null_mut();
-    let rc = unsafe {
-        spdk_nvmf_request_get_bdev(1, req, &mut bdev, &mut desc, &mut ch)
-    };
-    if rc != 0 {
-        /* No bdev found for this namespace. Continue. */
-        debug!("no bdev found");
-        return -1;
-    }
-
-    let bd = Bdev::checked_from_ptr(bdev).unwrap();
-    if bd.driver() == nexus::NEXUS_MODULE_NAME {
-        // Received command on a published Nexus
-        set_snapshot_time(unsafe { &mut *spdk_nvmf_request_get_cmd(req) });
-        unsafe {
-            spdk_nvmf_bdev_ctrlr_nvme_passthru_admin(bdev, desc, ch, req, None)
-        }
-    } else if let Ok(lvol) = Lvol::try_from(bd) {
-        // Received command on a shared replica (lvol)
-        let cmd = unsafe { spdk_nvmf_request_get_cmd(req) };
-        let snapshot_time = unsafe {
-            nvme_cmd_cdw10_get_val(cmd) as u64
-                | (nvme_cmd_cdw11_get_val(cmd) as u64) << 32
-        };
-        let snapshot_name =
-            Lvol::format_snapshot_name(&lvol.name(), snapshot_time);
-        let nvmf_req = NvmfReq(NonNull::new(req).unwrap());
-        // Blobfs operations must be on md_thread
-        Reactors::master().send_future(async move {
-            lvol.create_snapshot(&nvmf_req, &snapshot_name).await;
-        });
-        1 // SPDK_NVMF_REQUEST_EXEC_STATUS_ASYNCHRONOUS
-    } else {
-        debug!("unsupported bdev driver");
-        -1
-    }
-}
-
-pub fn create_snapshot(lvol: Lvol, cmd: &spdk_nvme_cmd, io: *mut spdk_bdev_io) {
-    let snapshot_time = unsafe {
-        nvme_cmd_cdw10_get_val(cmd) as u64
-            | (nvme_cmd_cdw11_get_val(cmd) as u64) << 32
-    };
-    let snapshot_name = Lvol::format_snapshot_name(&lvol.name(), snapshot_time);
-    // Blobfs operations must be on md_thread
-    Reactors::master().send_future(async move {
-        lvol.create_snapshot_local(io, &snapshot_name).await;
-    });
-}
-
-/// Register custom NVMe admin command handler
-pub fn setup_create_snapshot_hdlr() {
-    unsafe {
-        spdk_nvmf_set_custom_admin_cmd_hdlr(
-            nvme_admin_opc::CREATE_SNAPSHOT,
-            Some(nvmf_create_snapshot_hdlr),
-        );
-    }
-}
+// //! Handlers for custom NVMe Admin commands
```

Comment:
Do we need to comment all the code out ? If it's not used we can mark it as accordingly to keep linter happy and just keep it as it is.